tests/system/src/test/java/test/robot/javafx/stage/StageAttributesTest.java

```diff
@@ -299,12 +298,138 @@ public void testFullScreenStageBeforeShow() throws InterruptedException {
 
         // wait a little bit between getColor() calls - on macOS the below one
         // would fail without this wait
-        sleep(100);
+        sleep(SHORT_WAIT);
 
         runAndWait(() -> {
             // top left corner (plus some tolerance) should NOT show decorations
             Color color = getColor((int)topStage.getX() + 5, (int)topStage.getY() + 5);
             assertColorEquals(TOP_COLOR, color, TOLERANCE);
         });
     }
-}
+
+    @ParameterizedTest(name = PARAMETERIZED_TEST_DISPLAY)
+    @EnumSource(names = {"DECORATED", "UNDECORATED"})
+    void testStageStatePrecedenceOrderIconifiedMaximizedBeforeShow(StageStyle stageStyle) throws InterruptedException {
```

Comment:
The precedence rules written into the spec are very problematic and it would be a bear to implement them. For example, if you attempt to maximize an iconified stage on Windows it will de-iconify. Making it stay iconified and maximize later would take real effort. Maximizing an iconified stage on Mac does nothing (glass loses the maximized state) and fixing that would also take a lot of work. I really think our time would be better spent leaving things as-is and updating the spec.

I don't think the API calls for maximizing and iconifying the stage are useful to begin with. These are features provided to users so they can manager their desktop real estate. There's not a lot of compelling use cases for apps to do this.

Reply:
I agree we should do an update on the documentation and accept that platforms differ on some aspects (that are acceptable) such as this one.

Maximize maybe there's the case to start maximized to cover the screen for single-application use machines (common in enterprise software). I don't see the use for iconify.

I have removed this test and the related ones that test state precedence order.